### PR TITLE
[Documentation] Tutorials/how to guides for all features

### PR DESCRIPTION
**Pull request by Eduardodudu**
*Have you read the [Contributing Guidelines](https://github.com/Appsilon/.github/blob/main/CONTRIBUTING.md)?*

Issue - [Documentation] Tutorials/how to guides for all features](https://github.com/Appsilon/reactable.extras/issues/33)

## Changes description

1. Adjusted Readme to format and similar of other packages
2. Created a simple example to share on Readme
3. Created the tutorials tab and adjusted both `server side processing` and `custom-inputs` vignettes to tutorial
4. Created a new vignette tutorial `Combine reactable functionalities`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,10 +5,21 @@
 <!-- badges: start -->
 [![CRAN status](https://www.r-pkg.org/badges/version/reactable.extras)](https://cran.r-project.org/package=reactable.extras)
 [![R-CMD-check](https://github.com/Appsilon/reactable.extras/workflows/R-CMD-check/badge.svg)](https://github.com/Appsilon/reactable.extras/actions/workflows/main.yml)
+![Codecov test coverage](https://codecov.io/gh/Appsilon/reactable.extras/branch/main/graph/badge.svg)
 <!-- badges: end -->
 
+`reactable.extras` is an R package that enhances the functionality of the `reactable` package in Shiny applications. Reactable tables are interactive customizable, and `reactable.extras` extend their capabilities, allowing you to create dynamic and interactive data tables with ease.
```
Comment (andyquinterom) [inline README.md:11]: Add a hyperlink to reactable?

File: vignettes/tutorial/custom-inputs.rmd [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            "dropdown",
+            unique(df$Type),
+            class = "dropdown-extra"
+          )
+        ),
+        Model = colDef(
+          header = tooltip_extra("tooltip", content = "Model input"),
+          cell = text_extra(
+            "text"
+          )
```
Comment (andyquinterom) [inline vignettes/tutorial/custom-inputs.rmd:227]: The first argument "tooltip" on the `tooltip_extra` function is not necesarry. We should remove it.

File: vignettes/tutorial/custom-inputs.rmd [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+      )
+    })
+  }
+)
+
+```
+
+![](images/custom-inputs.gif)
+
+Example of saving the state when changing the page:
```
Comment (andyquinterom) [inline vignettes/tutorial/custom-inputs.rmd:140]: Do we still need this example?

**Review by andyquinterom — CHANGES_REQUESTED** [review]
I think it's all good. I would only update the example with the tooltips plus some extra comments I left.

File: vignettes/tutorial/custom-inputs.rmd (hunk 2 above)
Comment (Eduardodudu) [reply vignettes/tutorial/custom-inputs.rmd:227]: Updated! I noticed the background is white, but for general usage I think it's better

**Review by andyquinterom — APPROVED** [review]

**Pull request merged**